private/buf/bufformat/formatter.go

```diff
@@ -0,0 +1,921 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufformat
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/jhump/protocompile/ast"
+)
+
+// formatter writes an *ast.FileNode as a .proto file.
+type formatter struct {
```

Comment:
I might record a top-level `err` parameter, and add to it as you go using `multierr.Append`, and return it from `Run`. We might want to record errors in places such as `Out` if you go negative